### PR TITLE
fix(interactions): Press up event

### LINKED ISSUE #5290: [usePress] Hook doesn't call the onPress function when pressing the element at the corner
### Provide a general summary of the issue here

When tapping the corners of certain elements, the onPress function does not get triggered. This issue is observed on select mobile devices and can be replicated using Chrome Dev Tools by selecting Responsive > Mobile Touch. Multiple NextUI components that rely on the `usePress` hook are impacted by this behavior.

Video (Check last part)

https://github.com/adobe/react-spectrum/assets/30373425/6845dd0b-8e41-4911-a9f6-d3e281b0790a

### 🤔 Expected Behavior?

Call `onPress` function when pressing the element at its corners.

### 😯 Current Behavior

The `onPress` is not called when pressing any element/component that use the `usePress` under the hook.

### 💁 Possible Solution

I'm preparing a PR for this, we already fixed it in a local copy, but for us is better is this change is in the react-spectrum package.

### 🔦 Context

_No response_

### 🖥️ Steps to Reproduce

- Go to https://react-spectrum.adobe.com/react-aria/Button.html
- Open Chrome/Brave Dev Tools, Toggle device toolbar 
- Select Dimensions = Responsive
- Select "Mobile" 

### Version

3.19.1

### What browsers are you seeing the problem on?

Chrome, Microsoft Edge

### If other, please specify.

_No response_

### What operating system are you using?

macOS

### 🧢 Your Company/Team

NextUI

### 🕷 Tracking Issue

https://github.com/nextui-org/nextui/issues/1680
https://github.com/nextui-org/nextui/issues/1634
https://github.com/nextui-org/nextui/issues/1632
https://github.com/nextui-org/nextui/issues/1538

### PR DESCRIPTION
**Pull request by jrgarciadev**
Closes #5290 

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 🧢 Your Project:

https://github.com/nextui-org/nextui

### CODE REVIEW

File: packages/@react-aria/interactions/src/usePress.ts [hunk 1]
```diff
@@ -466,7 +466,7 @@ export function usePress(props: PressHookProps): PressResult {
           }
         } else if (state.isOverTarget) {
           state.isOverTarget = false;
-          triggerPressEnd(createEvent(state.target, e), state.pointerType, false);
+          triggerPressEnd(createEvent(state.target, e), state.pointerType);
```
Comment (devongovett) [inline packages/@react-aria/interactions/src/usePress.ts:469]: Won't this mean we trigger onPress when the user just presses down and then drags their mouse off without releasing? I think that's what the tests are catching as well.

Comment (jrgarciadev) [reply packages/@react-aria/interactions/src/usePress.ts:469]: I don't think so, here's a video, I pressed down the NextUI Button (custom usePress) and the `onPress` is not triggered when dragging without release, it seems to have the same behaviour.
 
https://github.com/adobe/react-spectrum/assets/30373425/a74baf7a-6192-4b5a-bfb6-56eeab9abd83

Here's the version with custom `usePress`:
https://nextui-storybook-v2-git-feat-autocomplete-nextui-org.vercel.app/?path=%2Fstory%2Fcomponents-button--with-state&vercelToolbarCode=e1L8-1_7if1HslV

Here's the version with react-aria `usePress`:
https://storybook.nextui.org/?path=/story/components-button--with-state

I don't see any difference

Comment (snowystinger) [reply packages/@react-aria/interactions/src/usePress.ts:469]: I'm not sure what's going on in nextui, but I just tried out this branch locally with our Button story in storybook and `press` action is fired if I mouse down and drag off the button.

Comment (jrgarciadev) [reply packages/@react-aria/interactions/src/usePress.ts:469]: Here's the code I implemented on NextUI https://github.com/nextui-org/nextui/pull/1797/files#diff-50bba2c230beaca7e521bcf1b5c27aba093884f9d15ee05011468abd71c7eda2

Comment (devongovett) [reply packages/@react-aria/interactions/src/usePress.ts:469]: could you do a diff between your version of usePress and the one in this PR to see what the difference is?

Comment (snowystinger) [reply packages/@react-aria/interactions/src/usePress.ts:469]: Hey, I did a diff of the file you linked to and our code. I do not see the change you're proposing in this PR in that diff...
From what I can tell, you meant to put it one more function down, in `onPointerUp`?

**Comment by snowystinger** [comment]
Hey! Just checking in if you still need this?

**Pull request closed without merging**